tests/v2/test_waku_noise.nim

```diff
@@ -157,4 +158,180 @@ procSuite "Waku Noise":
+    var decrypted: seq[byte] = decryptWithAd(cipherState, ad, ciphertext)
+
+    # We check if encryption and decryption are correct and that nonce correctly increased after decryption
+    check:
+      getNonce(cipherState) == nonce + 1 
+      plaintext == decrypted
+
+  test "Noise State Machine: Symmetric State primitives":
+
+    # We select one supported handshake pattern and we initialize a symmetric state
+    var 
+      hsPattern = NoiseHandshakePatterns["XX"]
+      symmetricState: SymmetricState =  SymmetricState.init(hsPattern)
+
+    # We get all the Symmetric State field
+    # cs : Cipher State
+    # ck : chaining key
+    # h : handshake hash
+    var
+      cs = getCipherState(symmetricState)
+      ck = getChainingKey(symmetricState)
+      h = getHandshakeHash(symmetricState)
+
+    ########################################
+    # mixHash
+    ########################################
+
+    # We generate a random byte sequence and execute a mixHash over it
+    mixHash(symmetricState, randomSeqByte(rng[], rand(1..128)))
+
+    # mixHash changes only the handshake hash value of the Symmetric state
+    check:
+      cs == getCipherState(symmetricState)
+      ck == getChainingKey(symmetricState)
+      h != getHandshakeHash(symmetricState)
+
+    # We update test values
+    h = getHandshakeHash(symmetricState)
+
+    ########################################
+    # mixKey
+    ########################################
+
+    # We generate random input key material and we execute mixKey
+    var inputKeyMaterial = randomChaChaPolyKey(rng[])
+    mixKey(symmetricState, inputKeyMaterial)
+
+    # mixKey changes the Symmetric State's chaining key and encryption key of the embedded Cipher State 
+    # It further sets to 0 the nonce of the embedded Cipher State
+    check:
+      getKey(cs) != getKey(getCipherState(symmetricState))
+      getNonce(getCipherState(symmetricState)) == 0.uint64
+      cs != getCipherState(symmetricState)
+      ck != getChainingKey(symmetricState)
+      h == getHandshakeHash(symmetricState)
+
+    # We update test values
+    cs = getCipherState(symmetricState)
+    ck = getChainingKey(symmetricState)
+
+    ########################################
+    # mixKeyAndHash
+    ########################################
+
+    # We generate random input key material and we execute mixKeyAndHash
+    inputKeyMaterial = randomChaChaPolyKey(rng[])
+    mixKeyAndHash(symmetricState, inputKeyMaterial)
+
+    # mixKeyAndHash executes a mixKey and a mixHash using the input key material
+    # All Symmetric State's fields are updated
+    check:
+      cs != getCipherState(symmetricState)
+      ck != getChainingKey(symmetricState)
+      h != getHandshakeHash(symmetricState)
+
+    # We update test values
+    cs = getCipherState(symmetricState)
+    ck = getChainingKey(symmetricState)
+    h = getHandshakeHash(symmetricState)
+
+    ########################################
+    # encryptAndHash and decryptAndHash
```

Comment:
Other things to consider for the tests is the edge cases and unhappy paths, like when the nonce reaches the max value.

Reply:
Addressed in https://github.com/status-im/nwaku/pull/933/commits/4ae4364845643190a3af13b3f15fd4afb6bcf906